zerver/webhooks/github/view.py

```diff
@@ -198,85 +199,90 @@ def get_deployment_body(helper: Helper) -> str:
 def get_change_deployment_status_body(helper: Helper) -> str:
     payload = helper.payload
     return "Deployment changed status to {}.".format(
-        payload["deployment_status"]["state"],
+        payload["deployment_status"]["state"].tame(check_string),
     )
 
 
 def get_create_or_delete_body(helper: Helper, action: str) -> str:
     payload = helper.payload
-    ref_type = payload["ref_type"]
+    ref_type = payload["ref_type"].tame(check_string)
     return "{} {} {} {}.".format(
         get_sender_name(payload),
         action,
         ref_type,
-        payload["ref"],
+        payload["ref"].tame(check_string),
     ).rstrip()
 
 
 def get_commit_comment_body(helper: Helper) -> str:
     payload = helper.payload
     comment = payload["comment"]
-    comment_url = comment["html_url"]
+    comment_url = comment["html_url"].tame(check_string)
     commit_url = comment_url.split("#", 1)[0]
     action = f"[commented]({comment_url})"
     return get_commits_comment_action_message(
         get_sender_name(payload),
         action,
         commit_url,
-        comment.get("commit_id"),
-        comment["body"],
+        comment.get("commit_id").tame(check_string),
```

Comment:
`foo.get("bar")` returns `str | None`, not `str`. Unfortunately, we have a bunch of existing code that uses `foo.get("bar")` when it really means `foo["bar"]`, so some investigation of the specific webhook is required to determine whether the right fix is

```python
foo["bar"].tame(check_string)  # for a string that’s really required
foo.get("bar").tame(check_none_or(check_string))  # for a string that be missing
```

There are many other instances of this.

Reply:
Yeap I noticed that `foo.get("bar")` returns `str | None`. So far I have been working on the assumption that if the code following `foo.get("bar")` does not explicitly handle the case when `foo.get("bar")` returns `None`, then they expect `bar` to be a required value. Together with the test cases that test on the different fixtures, I was hoping then would be enough to cover all grounds. But there might still be some cases that slip through the crack - where a required value is actually nullable. How else could we investigate the webhook to catch these cases?

Reply:
Yeap I noticed that `foo.get("bar")` returns `str | None`. So far I have been working on the assumption that if the code following `foo.get("bar")` does not explicitly handle the case when `foo.get("bar")` returns None, then they expect bar to be a required value. Together with the test cases that test on the different fixtures, I was hoping then would be enough to cover all grounds. But there might still be some cases that slip through the crack - where a required value is actually nullable. How else could we investigate the webhook to catch these cases?

Reply:
Usually by finding the upstream documentation for the webhook, e.g. [here](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#commit_comment) in this case. This isn’t actually explicit about whether a commit comment is required to have a `commit_id`, but it seems like a good assumption.

My point is that, whenever you make this assumption, you should also replace the ambiguous `comment.get("commit_id")` with an assertive `comment["commit_id"]`, so that violations of the assumption fail early.

Reply:
@andersk well noted!